DOC-1433: Add Cloud API examples for managing a subscription in API "Create and edit subscriptions" page (#2155)
RedisLabs/redislabs-docs: content/rc/api/examples/manage-subscriptions.md

         /rc/api/examples/manage-subscriptions.md
---

> The Redis Enterprise Cloud REST API allows you to create and manage a subscription.

Reviewer: I would avoid "allows you to". You could either start with the verb: 
Create and manage your subscription with the Redis Enterprise Cloud REST API. 
or use "lets" 
The Redis Enterprise Cloud REST API lets you create and manage your subscription.

}
```

> To use the sample JSON document in your own account, modify these parameters:

Reviewer: I would start this with the action: 
Modify the [following/these] parameters in the sample JSON document to create a subscription on your own account.

> Modify the following parameters in the sample JSON document to create a subscription on your own account.

Reviewer 2: ```suggestion
Modify the following parameters in the sample JSON document to create a subscription on your own account:
```

> - **`paymentMethodId`** - Specify a payment method that is connected to your account.

Reviewer: I would say either: 
...connected to your account. 
or
...for your account.
"that is connected to" is too many words

    Use `GET /payment-methods` to find a payment method ID.

>     You do not need to pass this field in your API request if you subscribed to Redis Enterprise Cloud through GCP Marketplace.

Reviewer: ```suggestion
    You don't need to pass this field in your API request if you subscribed through GCP Marketplace.
```

> - **`cloudAccountId`** - Set a cloud account ID that is connected to your account.

Reviewer: "that is connected to" -> same comment as above

>     You can look up cloud account IDs using `GET /cloud-accounts`. To use internal resources, set it to `"cloudAccountId": 1`.

Reviewer: ```suggestion
    To list cloud account IDs, use `GET /cloud-accounts`. 
    To use internal resources, set `cloudAccountID` to`"cloudAccountId": 1`.
```

>     If you subscribed to Redis Enterprise Cloud through GCP Marketplace, use `1` for this field.

Reviewer: ```suggestion
    If you subscribed through GCP Marketplace, use `1` for this field.
```

> The request JSON body contains two primary segments: subscription specification and databases specification. When you create a subscription, you must specify one or more databases in the "`databases`" array of the above JSON file.

Reviewer: ```suggestion
The request JSON body contains two segments: subscription specification and databases specification. When you create a subscription, you must specify one or more databases in the "`databases`" array.
```

You can [copy-and-paste]({{< relref  "/rc/api/get-started/use-rest-api.md#swagger-user-interface" >}}) the contents of the JSON file into the `POST /subscriptions` operation in the [Swagger UI](https://api.redislabs.com/v1/swagger-ui.html).

{{< note >}}
The Swagger UI generates default JSON examples for `POST` and `PUT` operations. You can reference these examples and modify them to fit your specific needs and account settings. The examples will fail if used as-is.
{{< /note >}}

> The response body contains the `taskId` of the task that is creating the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.

Reviewer: ```suggestion
The response body contains the `taskId` for the task creating the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.
```

## Update a subscription

Use `PUT /v1/subscriptions/<subscriptionId>` to update a subscription.

```sh
PUT "https://[host]/v1/subscriptions/<subscriptionId>"
{
    "name": "new-subscription-name",
    "paymentMethodId": <payment_id>
}
```

You can only change the following settings with this endpoint:
- **`name`** - Specify a new name for your subscription.

> - **`paymentMethodId`** - Specify a different payment method that is defined for your account.

Reviewer: ```suggestion
- **`paymentMethodId`** - Specify a different payment method for your account.
```

>     You can look up a payment method identifier using the `GET /payment-methods` API operation.

Reviewer: To find the IDs for payment methods connected to your account, use he `GET /payment-methods` API operation.

> The response body contains the `taskId` of the task that is updating the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.

Reviewer: ```suggestion
The response body contains the `taskId` for the task updating the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.
```

## Delete a subscription

Use `DELETE /v1/subscriptions/<subscriptionId>` to delete a subscription.

```sh
DELETE "https://[host]/v1/subscriptions/<subscriptionId>"
```

> The response body contains the `taskId` for the task deleting the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.

Reviewer 2: ```suggestion
The response body contains the `taskId` for the task that deletes the subscription. You can use `GET /v1/tasks/<taskId>` to track the task's status.
```

> The response body contains the `taskId` of the task that is deleting the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.

Reviewer: ```suggestion
The response body contains the `taskId` for the task deleting the subscription. You can use `GET /v1/tasks/<taskId>` to track the status of this task.
```
